elastic/kibana
# [Fleet] Rollover data streams when package w/ TSDB setting changed is installed

## Summary

Fixes https://github.com/elastic/kibana/issues/157345

When a package with a changed `index.mode` or `source.mode` setting is installed, Fleet will now automatically perform a rollover to ensure the correct setting is present on the resulting backing index.

There is an issue with Elasticsearch wherein toggling these settings back and forth will incur a backing index range overlap error. See https://github.com/elastic/elasticsearch/issues/96163. 

To test
1. Install the `system` integration at version `1.28.0`
2. Create an integration policy for the `system` integration (a standard default agent policy will do)
3. Enroll an agent in this policy, and allow it to ingest some data
4. Confirm that there are documents present in the `metrics-system.cpu-default` data stream, and note its backing index via Stack Management
5. Create a new `1.28.1` version of the `system` integration where `elasticsearch.index_mode: time_series` is set and install it via `elastic-package install --zip`
6. Confirm that a rollover occurs and the backing index for the `metrics-system.cpu-default` data stream has been updated

### Checklist

Delete any items that are not applicable to this PR.
